### PR TITLE
CMake: Disable VST (GTK) when gdk-x11-3.0 isn't found [wayland]

### LINKED ISSUE #614: [Wayland] VST depends on gdk/gdkx.h (GDK X11)
### Guidelines

- [X] I have read the guidelines.

### Version/Commit hash

01f22e8acb65b978181055cb6009093f5f6a3da4

### Describe the bug.

While building tenacity in a libX11-free system I had to reconfigure it and disable `use_vst` as it uses `gdk/gdkx.h`, which is for X11.

### Expected behavior

I think there should either be a test into the buildsystem for `gdk-x11-3.0` (pkg-config name) before allowing VST; or if possible removing the dependency on gdkx.

### OS

Gentoo w/ USE=-X

### Additional context

_No response_

### This issue is not a duplicate

- [X] I have confirmed this issue isn't a duplicate.

### PR DESCRIPTION
**Pull request by lanodan**
The VST plugin done via GTK for non-apple & non-windows pulls <gdk/gdkx.h>,
which is specific to X11 and thus isn't available in a pure-wayland system.

This disables it automatically.

Closes: https://github.com/tenacityteam/tenacity/issues/614

<details>
<summary>Checklist</summary>

- [x] I have signed off my commits using `-s` or `Signed-off-by`\* (See: [Contributing § DCO](https://github.com/tenacityteam/tenacity/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code\*
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving\*
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"\*

\* indicates required

</details>

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -58,8 +58,13 @@ cmd_option( use_ladspa "Use LADSPA plug-in support [ON, OFF]" ON )
 set( USE_LADSPA ${use_ladspa} CACHE INTERNAL "" )
 
 # Handle VST option
-cmd_option( use_vst "Use VST2 plug-in support [ON, OFF]" ON )
-set( USE_VST ${use_vst} CACHE INTERNAL "" )
+if( NOT CMAKE_SYSTEM_NAME MATCHES "Darwin|Windows" )
+  pkg_check_modules(GDK3_X11 gdk-x11-3.0)
+  cmake_dependent_option( use_vst "Use VST2 plug-in support [ON, OFF]" ON "GDK3_X11_FOUND" OFF)
+  set( USE_VST ${use_vst} CACHE INTERNAL "" )
```
Comment (Be-ing) [inline src/CMakeLists.txt]:
```suggestion
  cmake_dependent_option( USE_VST "Use VST2 plug-in support [ON, OFF]" ON "GDK3_X11_FOUND" OFF)
```
I don't think the intermediate lowercase variable is needed.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -58,8 +58,13 @@ cmd_option( use_ladspa "Use LADSPA plug-in support [ON, OFF]" ON )
 set( USE_LADSPA ${use_ladspa} CACHE INTERNAL "" )
 
 # Handle VST option
-cmd_option( use_vst "Use VST2 plug-in support [ON, OFF]" ON )
-set( USE_VST ${use_vst} CACHE INTERNAL "" )
+if( NOT CMAKE_SYSTEM_NAME MATCHES "Darwin|Windows" )
+  pkg_check_modules(GDK3_X11 gdk-x11-3.0)
```
Comment (Be-ing) [inline src/CMakeLists.txt]: This shows up in a different part of the configure output from the rest of the dependency checks. Could you move this up where the `find_package` calls and other options are?

Comment (Be-ing) [reply src/CMakeLists.txt]: Oh oops. I just noticed there is a separate chunk of code above which handles USE_VST. Could you replace that with this?

**Comment by Be-ing** [comment]
Thanks!

File: src/CMakeLists.txt (hunk 2 above)
Comment (lanodan) [reply src/CMakeLists.txt]: Not really sure what you meant there but I guess moving `USE_VST` close to `VST2` makes sense.

btw removing the `use_vst` option breaks the option name, which I think is okay given no release has been done yet but while we're at it, I noticed that both `use_vst` and `VST2` are about VST2, shouldn't they be either on the same option or be more clear on their descriptions?

Comment (Be-ing) [reply src/CMakeLists.txt]: > removing the use_vst option breaks the option name

Yes, that's intentional. I got rid of those weird lowercase options with annoying superfluous prefixes.

>  shouldn't they be either on the same option

Yes, could you make them the same option named `VST2`? If this option is enabled, `set(USE_VST TRUE)` so other parts of the CMake scripts that depend on that continue to work.

**Comment by n0toose** [comment]
Hi, thanks for your contribution (huge fan of your overlay and Pleroma BTW), I'll also give this a check and approve it after @Be-ing gives it the go! ^-^

**Comment by emabrey** [comment]
Can you sign off that newest commit (`git commit --amend --signoff --no-edit`)?

File: CMakeLists.txt [hunk 3]
```diff
@@ -788,17 +788,18 @@ else()
...
 if( NOT CMAKE_SYSTEM_NAME MATCHES "Darwin|Windows" )
   find_package( GLIB REQUIRED )
   find_package( GTK 3.0 REQUIRED )
+  pkg_check_modules(GDK3_X11 gdk-x11-3.0)
+endif()
+
+if( NOT CMAKE_SYSTEM_NAME MATCHES "Darwin|Windows" )
+  cmake_dependent_option( VST2 "Use VST2 plug-in support [ON, OFF]" ON "GDK3_X11_FOUND" OFF)
+else()
+  cmd_option( VST2 "Use VST2 plug-in support [ON, OFF]" ON)
```
Comment (Be-ing) [inline CMakeLists.txt]:
```suggestion
  option( VST2 "Use VST2 plug-in support" ON)
```
`cmd_option` is some weird overcomplicated thing from Audacity. Just use standard CMake `option`

**Comment by Be-ing** [comment]
Could you update BUILDING.md to mention this requirement for the VST2 option?

**Review by Be-ing — APPROVED** [review]
Thank you.

**Review by n0toose — APPROVED** [review]
Looks good to me. Thanks much!

**Pull request merged**